src/main/java/io/tackle/pathfinder/services/AssessmentSvc.java

```diff
@@ -56,33 +57,47 @@ public AssessmentHeaderDto createAssessment(@NotNull Long applicationId) {
     }
 
     @Transactional
-    public Assessment copyQuestionnaireIntoAssessment(Long assessmentId, Long questionnaireId) {
-        Questionnaire questionnaire = Questionnaire.findById(questionnaireId);
-        Assessment assessment = Assessment.findById(assessmentId);
-
-        AssessmentQuestionnaire assessQuestionnaire = AssessmentQuestionnaire.builder().name(questionnaire.name)
-                .questionnaire(questionnaire).assessment(assessment).languageCode(questionnaire.languageCode).build();
+    public Assessment copyQuestionnaireIntoAssessment(Assessment assessment, Questionnaire questionnaire) {
+
+        AssessmentQuestionnaire assessQuestionnaire = AssessmentQuestionnaire.builder()
+                .name(questionnaire.name)
+                .questionnaire(questionnaire)
+                .assessment(assessment)
+                .languageCode(questionnaire.languageCode)
+                .build();
         assessQuestionnaire.persist();
 
         assessment.assessmentQuestionnaire = assessQuestionnaire;
 
         assessment.assessmentQuestionnaire.categories = new ArrayList<AssessmentCategory>();
         for (Category category : questionnaire.categories) {
-            AssessmentCategory assessmentCategory = AssessmentCategory.builder().name(category.name)
-                    .order(category.order).questionnaire(assessment.assessmentQuestionnaire )
-                    .questions(new ArrayList<>()).build();
+            AssessmentCategory assessmentCategory = AssessmentCategory.builder()
+                    .name(category.name)
+                    .order(category.order)
+                    .questionnaire(assessment.assessmentQuestionnaire )
```

Comment:
Why don't we instantiate the questions here any more?

Reply:
Is it already instantiated in the Entity